File: apigateway_getting_started/create_api_client_deploy_stage.js
import { APIGatewayClient, CreateDeploymentCommand } from "@aws-sdk/client-api-gateway";
(async function () {
    const apig = new APIGatewayClient({ region: "us-east-1" });
    const command = new CreateDeploymentCommand({
        restApiId: 'abcd1234',
        stageName: 'test',
        stageDescription: 'test deployment'
    });
    try {
        const results = await apig.send(command)
        console.log("Deploying API succeeded\n", results)
    } catch (err) {
        console.log("Deploying API failed:\n", err)
    }
})();

File: apigateway_getting_started/create_api_client_deploy_stage.py
import botocore
import boto3
import logging

logger = logging.getLogger()
apig = boto3.client("apigateway")

try:
    result = apig.create_deployment(
        restApiId="ieps9b05sf",
        stageName="test",
        stageDescription="my test stage",
    )
except botocore.exceptions.ClientError as error:
    logger.exception("Error deploying stage  %s.", error)
    raise
print("Deploying API succeeded")
print(result)
